src/os/bluestore/BlueRocksEnv.cc

```diff
@@ -429,7 +429,10 @@ rocksdb::Status BlueRocksEnv::GetChildren(
   std::vector<std::string>* result)
 {
   result->clear();
-  int r = fs->readdir(dir, result);
+  // dir may contain trailing /
+  std::string dir_only, empty;
+  split(dir, &dir_only, &empty);
```

Comment:
@TheJJ  this does not work if there is no trailing "/" in `dir` if we does not update the existing implementation of `split()`.  i update the corresponding commit in #39412 to use a different way for trimming the trailing "/".

since my changeset also includes a refactor to use `string_view` in BlueFS and BlueRocksEnv, either you need to rebase or i need to do so. so i think probably we can continue on out there. what do you think?

Reply:
The easiest way is probably if you just incorporate my change in your patches and we just close this pull :)

Reply:
yeah, and good news! the "make check" passed with your change (revised version)!